File: 2017-12-13/gopath/src/code/c-example/c-example.go
package main

import "fmt"

func main() {
	type foo struct {
		x int
		y uint
		t string // always utf-8
	}

	manyFoo := []foo{
		{1, 2, "foo"},
		{3, 5, "bar"},
	}

	// i := 0; i < len(manyFoo); i++
	for i, foo := range manyFoo {
		fmt.Printf("%v %v\n", i, foo)
	}
}

File: 2017-12-13/gopath/src/code/c-example/c-example.c
#include <stdio.h>

#define ARR_SIZE(__x) \
  sizeof(__x) / sizeof(__x[0])

int main(int argc, char *argv[]) {
  typedef struct {
    int x;
    unsigned int y;
    const char *t;
  } foo_t;

  foo_t many_foo[] = {
    {1, 2, "foo"},
    {3, 5, "bar"},
  };
  for (size_t i = 0; i < ARR_SIZE(many_foo); i++) {
    printf("%zu: {%d %d \"%s\"}\n", i,
           many_foo[i].x, many_foo[i].y, many_foo[i].t);
  }
  return 0;
}
